scene/resources/packed_scene.cpp

```diff
@@ -238,16 +238,38 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 
 					if (nprops[j].name & FLAG_PATH_PROPERTY_IS_NODE) {
 						uint32_t name_idx = nprops[j].name & (FLAG_PATH_PROPERTY_IS_NODE - 1);
+						const NodeData::Property &nprop = nprops[j];
+						const Variant &prop_variant = props[nprop.value];
+						const StringName &prop_name = snames[name_idx];
 						ERR_FAIL_UNSIGNED_INDEX_V(name_idx, (uint32_t)sname_count, nullptr);
-						if (Engine::get_singleton()->is_editor_hint()) {
-							// If editor, just set the metadata and be it
-							node->set(META_POINTER_PROPERTY_BASE + String(snames[name_idx]), props[nprops[j].value]);
+
+						if (prop_variant.get_type() == Variant::ARRAY) {
+							if (Engine::get_singleton()->is_editor_hint()) {
+								// If editor, just set the metadata and be it
+								node->set(prop_name, prop_variant);
+								continue;
+							}
```

Comment:
Why does it need to be special cased here? The only difference seems to be not using the `META_POINTER_PROPERTY_BASE` prefix, does it break Node arrays?

Reply:
This will assign the raw array of NodePaths. `META_POINTER_PROPERTY_BASE` doesn't seem to work here. The condition can't be moved elsewhere and removing it crashes, so it's likely the only way.